### PR TITLE
Fix/chip right icon margin

### PR DESCRIPTION
**Pull request by NitzanWix**
## Description
Added horizontal margins to the right icon in the chip component. The icon was really close to the label when rendering the icon without setting margins manually.

## Changelog
Chip - Added default horizontal margins to right icon.

## Additional info
WOAUILIB-3772

### CODE REVIEW

File: src/components/chip/index.tsx [hunk 1]
```diff
@@ -292,6 +292,13 @@ const Chip = ({
           return {
             marginRight: Spacings.s2
           };
+        case 'iconSource':
+          if (label || leftElement || rightElement) {
+            return {
+              marginLeft: Spacings.s1,
+              marginRight: Spacings.s2
+            };
```
Comment (ethanshar) [inline src/components/chip/index.tsx:300]: overall It looks much better. 
But I think it mostly solve the right icon
notice that the use can either pass `rightIconSource` or `iconSource` (rendered on the left)
I think those margins works good for right icons and maybe should be opposite for left icon

File: src/components/chip/index.tsx [hunk 2]
```diff
@@ -255,22 +255,25 @@ const Chip = ({
...
+                marginRight: Spacings.s3
+              };
+            }
+            if (rightIconSource || rightElement) {
+              return {
+                marginLeft: Spacings.s3,
+                marginRight: marginFromElement
+              };
+            }
           }
```
Comment (ethanshar) [inline src/components/chip/index.tsx:277]: A small suggestion for making this a little more readable
you can define two consts
```
const addLeftMargin = !!(iconSource || leftElement)
const addRightMargin = !!(rightIconSource || rightElement)
```
Then the conditions will much more readable. Also if there will new elements in either left or right we can simply add them to the const definition and that's it

Last, consider using ternary to make things a little shorter, for example 
```
const marginStyle = {
  marginHorizontal: addLeftMargin && addLeftMargin ? Spacings.1 : 0,
 marginLeft: ...
 marginRight: ... 
}
``

Comment (NitzanWix) [reply src/components/chip/index.tsx:277]: Did you mean
`marginHorizontal: addLeftMargin && addRightMargin ? Spacings.s1 : 0`?
About the last part. I don't really understand what you mean, If I use marginHorizontal and margin left or right in the same object marginHorizontal will always be overridden by the left and right margins. So I don't really know how to use them together in the same object.

File: expoDemo/package.json
Comment (ethanshar) [inline expoDemo/package.json:1]: Can you revert this file changes, it's not relevant to the PR

Comment (NitzanWix) [reply expoDemo/package.json:1]: Weird. Idk where it came from. Done

**Review by ethanshar — APPROVED** [review]

**Pull request merged**